controller/src/run.rs

```diff
@@ -14,6 +16,35 @@ use std::future::Future;
 use std::pin::Pin;
 use std::thread;
 
+/// Receive UpdateBackendStateMessages over core NATS, and turn them into
+/// BackendStateMessages over JetStream.
+pub async fn update_backend_state_loop(nc: TypedNats) -> NeverResult {
+    let mut sub = nc
+        .subscribe(UpdateBackendStateMessage::subscribe_subject())
+        .await?;
+    loop {
+        let msg = sub
+            .next()
+            .await
+            .ok_or_else(|| anyhow!("UpdateBackendStateMessage subscription ended."))?;
```

Comment:
what exactly happens if a subscription ends like this? There's a nats loop somewhere else that restarts all these loops on reconnect right?

Reply:
Currently, no. These subscriptions shouldn't end based on my understanding of NATS, and I want to know if it happens.